control-plane/stor-port/src/types/v0/store/nexus.rs

```diff
@@ -232,6 +255,13 @@ impl SpecTransaction<NexusOperation> for NexusSpec {
                 }
                 NexusOperation::AddChild(uri) => self.children.push(uri),
                 NexusOperation::RemoveChild(uri) => self.children.retain(|c| c != &uri),
+                NexusOperation::RebuildState(child_uri, stage) => {
+                    if let Some(stage) = stage {
```

Comment:
Do you intend to check `stage.is_some()` here?

Reply:
Mostly yes. If i used is_some() we get bool and will have to do unwrap while inserting to hashmap. with if let Some(stage) this does unwrap if its Some or else we just delete key from the hashmap.